File: src/main/java/net/tecdroid/core/Robot.java
package net.tecdroid.core;

import edu.wpi.first.wpilibj.DriverStation;
import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj2.command.Command;
import edu.wpi.first.wpilibj2.command.CommandScheduler;

public class Robot extends TimedRobot {
    private Command autonomousCommand;

    private final RobotContainer container;

    public Robot() {
        container = new RobotContainer();
    }

    @Override
    public void robotInit() {
        container.setup();
        DriverStation.silenceJoystickConnectionWarning(true);
    }

    @Override
    public void robotPeriodic() {
        CommandScheduler.getInstance()
                        .run();
    }

    @Override
    public void disabledInit() {
    }

    @Override
    public void disabledPeriodic() {
    }

    @Override
    public void disabledExit() {

    }

    @Override
    public void autonomousInit() {
        autonomousCommand = container.getAutonomousCommand();

        if (autonomousCommand != null) {
            autonomousCommand.schedule();
        }
    }

    @Override
    public void autonomousPeriodic() {
    }

    @Override
    public void teleopInit() {
        if (autonomousCommand != null) {
            autonomousCommand.cancel();
        }

    }

    @Override
    public void teleopPeriodic() {
        container.teleop();
    }

    @Override
    public void testInit() {
        CommandScheduler.getInstance()
                        .cancelAll();
    }

    @Override
    public void testPeriodic() {
    }

    @Override
    public void simulationInit() {
    }

    @Override
    public void simulationPeriodic() {
    }
}

File: src/main/java/net/tecdroid/core/Robot.kt
package net.tecdroid.core

import edu.wpi.first.wpilibj.DriverStation
import edu.wpi.first.wpilibj.TimedRobot
import edu.wpi.first.wpilibj2.command.Command
import edu.wpi.first.wpilibj2.command.CommandScheduler
import net.tecdroid.util.volts

class Robot : TimedRobot() {
    private val container = RobotContainer()
    private val autonomousCommand: Command
        get() = container.autonomousCommand

    override fun robotInit() {
        DriverStation.silenceJoystickConnectionWarning(true)
        container.limelightController.setThrottle(200)
    }

    override fun robotPeriodic() {
        CommandScheduler.getInstance().run()
        container.robotPeriodic()
    }

    override fun disabledInit() {
        CommandScheduler.getInstance().cancelAll()
        container.limelightController.setThrottle(200)
    }

    override fun disabledPeriodic() {
    }

    override fun disabledExit() {
    }

    override fun autonomousInit() {
        container.limelightController.setThrottle(0)
        container.autonomousInit()
        autonomousCommand.schedule()
    }

    override fun autonomousPeriodic() {
    }

    override fun teleopInit() {
        container.limelightController.setThrottle(0)
        container.teleopInit()
        if (autonomousCommand.isScheduled) autonomousCommand.cancel()
    }

    override fun teleopPeriodic() {
    }

    override fun testInit() {
        CommandScheduler.getInstance().cancelAll()
    }

    override fun testPeriodic() {
    }

    override fun simulationInit() {
    }

    override fun simulationPeriodic() {
    }
}
